src/cfnlint/rules/functions/FindInMapKeys.py

```diff
@@ -0,0 +1,90 @@
+  without restriction, including without limitation the rights to use, copy, modify,
+  merge, publish, distribute, sublicense, and/or sell copies of the Software, and to
+  permit persons to whom the Software is furnished to do so.
+
+  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+  INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+  PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+  HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+  OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+  SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+"""
+import six
+from cfnlint import CloudFormationLintRule
+from cfnlint import RuleMatch
+
+
+class FindInMapKeys(CloudFormationLintRule):
+    """Check if FindInMap values are correct"""
+    id = 'E1013'
+    shortdesc = 'FindInMap key\'s exist in the map'
+    description = 'Checks the keys in a FindInMap to make sure they exist. ' \
+                  'Check only if the Map Name is a string and if the key is a string.'
+    source_url = 'https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-findinmap.html'
+    tags = ['functions', 'findinmap']
+
+    def first_key(self, map_name, first_key, mappings, tree):
+        """ Check the validity of the first key """
+        matches = []
+        if isinstance(first_key, (six.string_types, int)):
+            if isinstance(map_name, (six.string_types)):
+                mapping = mappings.get(map_name)
+                if mapping:
+                    if not mapping.get(first_key):
+                        message = 'FindInMap first key "{0}" doesn\'t exist in map "{1}" at {3}'
+                        matches.append(RuleMatch(
+                            tree[:] + [1],
+                            message.format(first_key, map_name, first_key, '/'.join(map(str, tree)))))
+
+        return matches
+
+    def second_key(self, map_name, first_key, second_key, mappings, tree):
+        """ Check the validity of the first key """
+        matches = []
+        if isinstance(second_key, (six.string_types, int)):
+            if isinstance(map_name, (six.string_types)):
+                mapping = mappings.get(map_name)
+                if mapping:
+                    if isinstance(first_key, six.string_types):
+                        if mapping.get(first_key):
+                            # Don't double error if they first key doesn't exist
+                            if not mapping.get(first_key, {}).get(second_key):
+                                message = 'FindInMap second key "{0}" doesn\'t exist in map "{1}" under "{2}" at {3}'
+                                matches.append(RuleMatch(
+                                    tree[:] + [2],
+                                    message.format(second_key, map_name, first_key, '/'.join(map(str, tree)))))
```

Comment:
Since you have to validate the first key again here anyway, does it make sense to just pass both keys and validate them together in single function?

Reply:
Good question.  I could do that I suppose.  I was just trying to break it up for readability and to keep the functions small.